lib/dpl/provider/bintray.rb

```diff
@@ -0,0 +1,505 @@
+require 'json'
+require 'net/http'
+require "uri"
+require 'find'
+
+module DPL
+  class Provider
+    class Bintray < Provider
+      @test_mode = false
+
+      def check_auth
+      end
+
+      def needs_key?
+        false
+      end
+
+      attr_writer :test_mode
+
+      def init_from_args
+        @user = options[:user]
+        @key = options[:key]
+        @url = options[:url]
+        @file = options[:file]
+        @passphrase = options[:passphrase]
+        @dry_run = options[:dry_run]
+
+        if @user.nil?
+          abort("The 'user' argument is required")
+        end
+        if @key.nil?
+          abort("The 'key' argument is required")
+        end
+        if @file.nil?
+          abort("The 'file' argument is required")
+        end
+        if @url.nil?
+          @url = 'https://api.bintray.com'
+        end
+        if @dry_run.nil?
+          @dry_run = false
+        end
+      end
+
+      def read_descriptor
+        log "Reading descriptor file: #{@file}"
+        @descriptor = JSON.parse(File.read(@file))
```

Comment:
Using instance variables everywhere is generally not a good idea. Either use the attribute accessor (or reader), so that if one ever wants to change the way it is accessed it can be done in just one place.